Verify Kubernetes with Docker 1.7 RC

Looks like the 1.7 RC is out:

https://gist.github.com/vieux/42189ac13129efbadd15

Ok, I brought up an e2e cluster with docker 1.7.0-rc2, and here are the initial test result:

Summarizing 16 Failures:

[Fail] Density [BeforeEach] [Performance suite] should allow starting 3 pods per node 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/density.go:63

[Fail] Density [BeforeEach] [Performance suite] should allow starting 30 pods per node 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/density.go:63

[Fail] Services [It] should be able to create a functioning external load balancer 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:272

[Fail] Services [It] should correctly serve identically named services in different namespaces on different external IP addresses 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:796

[Fail] Secrets [It] should be consumable from pods 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/util.go:645

[Fail] Services [It] should be able to change the type and nodeport settings of a service 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:413

[Fail] Services [It] should release the load balancer when Type goes from LoadBalancer -> NodePort 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:542

[Fail] ResizeNodes [AfterEach] should be able to add nodes. 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/resize_nodes.go:241

[Fail] Services [It] should be able to create a functioning NodePort service 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:1032

[Fail] DNS [It] should provide DNS for services 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/dns.go:138

[Fail] Cluster level logging using Elasticsearch [It] should check that logs from pods on all nodes are ingested into Elasticsearch 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:148

[Fail] Networking [It] should function for intra-pod communication 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:136

[Fail] ResizeNodes [It] should be able to delete nodes. 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/resize_nodes.go:273

[Fail] SSH [It] should SSH to all nodes and run commands 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/ssh.go:76

[Fail] PreStop [It] should call prestop when killing a pod 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pre_stop.go:141

[Fail] Pod Disks [It] should schedule a pod w/ a readonly PD on two hosts, then remove both. 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pd.go:157

Ran 61 of 70 Specs in 7700.197 seconds
FAIL! -- 45 Passed | 16 Failed | 2 Pending | 7 Skipped --- FAIL: TestE2E (7700.63s)
FAIL

Of course some test failures are caused by previous failures, for example: #9345. But things look badly so far. Next step is debugging with one or two failures. Will update with more information later.

v1.7 has networking and volumes changes. So its required to probe the test
failures in detail. Thanks for picking this up Dawn.

On Fri, Jun 5, 2015 at 4:00 PM, Dawn Chen <notifications@github.com> wrote:

> Ok, I brought up an e2e cluster with docker 1.7.0-rc2, and here are the
> initial test result:
>
> Summarizing 16 Failures:
>
> [Fail] Density [BeforeEach] [Performance suite] should allow starting 3
> pods per node
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/density.go:63
>
> [Fail] Density [BeforeEach] [Performance suite] should allow starting 30
> pods per node
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/density.go:63
>
> [Fail] Services [It] should be able to create a functioning external load
> balancer
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:272
>
> [Fail] Services [It] should correctly serve identically named services in
> different namespaces on different external IP addresses
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:796
>
> [Fail] Secrets [It] should be consumable from pods
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/util.go:645
>
> [Fail] Services [It] should be able to change the type and nodeport
> settings of a service
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:413
>
> [Fail] Services [It] should release the load balancer when Type goes from
> LoadBalancer -> NodePort
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:542
>
> [Fail] ResizeNodes [AfterEach] should be able to add nodes.
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/resize_nodes.go:241
>
> [Fail] Services [It] should be able to create a functioning NodePort
> service
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:1032
>
> [Fail] DNS [It] should provide DNS for services
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/dns.go:138
>
> [Fail] Cluster level logging using Elasticsearch [It] should check that
> logs from pods on all nodes are ingested into Elasticsearch
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:148
>
> [Fail] Networking [It] should function for intra-pod communication
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:136
>
> [Fail] ResizeNodes [It] should be able to delete nodes.
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/resize_nodes.go:273
>
> [Fail] SSH [It] should SSH to all nodes and run commands
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/ssh.go:76
>
> [Fail] PreStop [It] should call prestop when killing a pod
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pre_stop.go:141
>
> [Fail] Pod Disks [It] should schedule a pod w/ a readonly PD on two hosts,
> then remove both.
> /go/src/
> github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pd.go:157
>
> Ran 61 of 70 Specs in 7700.197 seconds
> FAIL! -- 45 Passed | 16 Failed | 2 Pending | 7 Skipped --- FAIL: TestE2E
> (7700.63s)
> FAIL
>
> Of course some test failures are caused by previous failures, for example:
> #9345 <https://github.com/GoogleCloudPlatform/kubernetes/issues/9345>.
> But things look badly so far. Next step is debugging with one or two
> failures. Will update with more information later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/issues/8918#issuecomment-109472383>
> .
>

--- a/cluster/saltbase/salt/docker/init.sls
+++ b/cluster/saltbase/salt/docker/init.sls
@@ -99,15 +99,9 @@ net.ipv4.ip_forward:
 
 {% set storage_base='https://storage.googleapis.com/kubernetes-release/docker/' %}
 
-{% set override_deb='lxc-docker-1.6.0_1.6.0_amd64.deb' %}
-{% set override_deb_sha1='fdfd749362256877668e13e152d17fe22c64c420' %}
-{% set override_docker_ver='1.6.0' %}
-
-{% if grains.cloud is defined and grains.cloud == 'gce' %}
-{% set override_deb='' %}
-{% set override_deb_sha1='' %}
-{% set override_docker_ver='' %}
-{% endif %}
+{% set override_deb='lxc-docker-1.7.1_1.7.1_amd64.deb' %}
+{% set override_deb_sha1='81abef31dd2c616883a61f85bfb294d743b1c889' %}
+{% set override_docker_ver='1.7.1' %}
 
 {% if override_docker_ver != '' %}
 /var/cache/docker-install/{{ override_deb }}:
@@ -180,6 +174,9 @@ docker:
 {% endif %}
     - watch:
       - file: {{ environment_file }}
+{% if override_docker_ver != '' %}
+      - pkg: lxc-docker-{{ override_docker_ver }}
+{% endif %}
 {% if pillar.get('is_systemd') %}
       - file: {{ pillar.get('systemd_system_path') }}/docker.service
 {% endif %}
@@ -187,5 +184,4 @@ docker:
     - require:
       - pkg: lxc-docker-{{ override_docker_ver }}
 {% endif %}
-
 {% endif %} # end grains.os_family != 'RedHat'

--- a/cluster/saltbase/salt/docker/init.sls
+++ b/cluster/saltbase/salt/docker/init.sls
@@ -103,7 +103,19 @@ net.ipv4.ip_forward:
 {% set override_deb_sha1='81abef31dd2c616883a61f85bfb294d743b1c889' %}
 {% set override_docker_ver='1.7.1' %}
 
+# Comment out below logic for master branch, so that we can upgrade GCE cluster
+# to docker 1.7.1 by default.
+#
+# TODO(dchen1107): For release 1.1, we want to fall back to
+# ContainerVM installed docker by set override_deb, override_deb_sha1 and
+# override_docker_ver back to '' for gce cloud provider.
+
 {% if override_docker_ver != '' %}
+purge-old-docker-package:
+  pkg.removed:
+    - pkgs:
+      - lxc-docker-1.6.2
+
 /var/cache/docker-install/{{ override_deb }}:
   file.managed:
     - source: {{ storage_base }}{{ override_deb }}

